### ISSUE TITLE
Performance Drop When Using OBB

### ISSUE THREAD

**Issue by KHC1234** [ultralytics/ultralytics#18727]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have compared the performance of OBB and HBB for object detection using YOLOv8. Why do I find out that the performance of my implementation of OBB is significantly worse than HBB (both precision and recall)? The precision is around 0.57 and recall around 0.6 for OBB while precision and recall are around 0.75 for HBB. OBB should in principle have better performance than HBB, isn't it?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @KHC1234, thank you for bringing up this interesting question regarding OBB vs. HBB performance in YOLOv8 🚀!

We recommend reviewing the Ultralytics [Documentation](https://docs.ultralytics.com) for more insights. There, you'll find helpful details and tips about [Model Training](https://docs.ultralytics.com/guides/model-training-tips/) that might inform your setup and results.

If this is related to a 🐛 bug, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) that includes:
1. The specific commands or scripts you used.
2. Details about your dataset (e.g., format, sample images, annotations, etc.).
3. Logs or outputs related to the training or validation process.

This will assist us in debugging the issue more efficiently.

For performance or custom training ❓ questions, it would also help to include:
- Dataset balance and characteristics, especially the annotations related to OBB and HBB.
- Training hyperparameters used in your experiments.
- Training and validation logs to compare metrics.

Don't forget, performance differences between OBB and HBB can occur based on various factors like annotation quality, the ratio of object sizes, or implementation differences. We also recommend double-checking the guidance in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure you have optimized your model training process.

## Upgrade

First, please make sure you are using the latest version of the `ultralytics` package to ensure you benefit from the latest updates and fixes. Upgrade your package by running:

```bash
pip install -U ultralytics
```

This ensures any improvements or bug fixes are included in your testing.

## Environments

YOLOv8 can work seamlessly across various platforms. You can test your setup or try reproducing your results in one of these verified environments to ensure consistency:
- **Notebooks** with free GPUs:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>  
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>  
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)).
- **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)).
- **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/))  
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Community Support

Feel free to connect with the Ultralytics community for further discussions or insights:
- Join us on [Discord](https://discord.com/invite/ultralytics) for real-time help 🎧.
- Participate in deeper discussions on [Discourse](https://community.ultralytics.com).
- Share and learn on the [Ultralytics Subreddit](https://reddit.com/r/Ultralytics).  

We love to see collaboration and shared knowledge within the community!

## Status

Ensure you're aware of the CI testing status via this badge:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

The badge status will indicate if all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests validate YOLO operation across various platforms and tasks.

---

⚠️ Please note, this is an automated response to assist you quickly. An Ultralytics engineer will review your issue and provide further assistance soon. Feel free to share additional details or updates in the meantime 😊!

**Comment by Y-T-G** [comment]
They are calculated differently.

> OBB should in principle have better performance than HBB, isn't it?

OBB requires both correct coordinates and angle, while HBB requires only correct coordinates. OBB is stricter so it should score less.

**Comment by KHC1234** [comment]
> They are calculated differently.
> 
> > OBB should in principle have better performance than HBB, isn't it?
> 
> OBB requires both correct coordinates and angle, while HBB requires only correct coordinates. OBB is stricter so it should score less.

Thanks for your reply. I have supplied additional coordinates and angles for OBB, but the performance statistics drop.

**Comment by Y-T-G** [comment]
It is stricter, so it would probably score less.

**Comment by KHC1234** [comment]
> It is stricter, so it would probably score less.

But I have looked at the output detection results. A number of objects correctly detected by HBB cannot be detected by OBB. So this is not only the score is lower, but some real outputs are lost.

**Comment by Y-T-G** [comment]
Are they overlapping?

**Comment by KHC1234** [comment]
> Are they overlapping?

Some missed outputs are not overlapping.

**Comment by KHC1234** [comment]
> You can try training for longer.

The learning curves indicate that the model starts overfitting.

**Comment by Y-T-G** [comment]
The best.pt is automatically saved, so you don't really have to worry about that.
